### PR TITLE
Add support for reading YAML files

### PR DESCRIPTION
**Pull request by iravid**
Resolves #379.

Uses snakeyaml-engine to add support for YAML 1.2 files as sources.

A possible follow-up to this would be support for multi-document yaml files.

### CODE REVIEW

**Comment by afsalthaj** [comment]
Looks like a clean solution and mapped nicely to PropertyTree.

**Comment by iravid** [comment]
Sure let me add such a test before you merge.

**Comment by hmemcpy** [comment]
Very much looking forward seeing this merged :)

File: yaml/src/test/scala/zio/config/yaml/YamlConfigSpec.scala [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        ConfigDescriptor.nested("sum") {
+          ConfigDescriptor.list {
+            (ConfigDescriptor.nested("A")(ConfigDescriptor.string("a")(A.apply, A.unapply)) orElseEither
+              ConfigDescriptor.nested("B")(ConfigDescriptor.boolean("b")(B.apply, B.unapply)))
+              .xmap(_.merge, (_: Sum) match {
+                case a: A => Left(a)
+                case b: B => Right(b)
+              })
+          }
+        }(Child.apply, Child.unapply)
```
Comment (afsalthaj) [inline yaml/src/test/scala/zio/config/yaml/YamlConfigSpec.scala:70]: Yeah, currently forming a sealed trait is ncier only when using `descriptor[Child]` through zio-config-magnolia, else having to do this verbose manual config

**Pull request merged**